### PR TITLE
test(ingest/bigquery): Add performance testing framework for bigquery usage

### PR DESCRIPTION
**Pull request by asikowitz**
Performance testing framework that contains:
- A simplified model of a data warehouse's metadata (tables, views, queries, etc.)
- A function for generating examples of such a model of different sizes
- A function for converting generic `Query`s into BigQuery `AuditEvent`s
- An unfinished test for testing bigquery usage performance

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/tests/performance/test_bigquery_usage.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    try:
+        root_logger.addHandler(stream_handler)
+        yield from set_log_level(root_logger, logging.ERROR)
+    finally:
+        root_logger.removeHandler(stream_handler)
+
+
+@pytest.fixture
+def report_log_level_info():
+    yield from set_log_level(report_logger, logging.INFO)
```
[the remark this answers is not in the document]
Comment (asikowitz) [reply metadata-ingestion/tests/performance/test_bigquery_usage.py:48]: I didn't see any error logs which made debugging these tests pretty annoying. Is this not usually the case and I'm doing something wrong? I also want to see INFO logs from the report class, so that when you run these tests, you can see how long each stage took

File: metadata-ingestion/tests/performance/test_bigquery_usage.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        num_operations=20000,
+        num_users=10,
+    )
+    events = generate_events(queries, projects, table_to_project, config=config)
+    events = list(events)
+    print(f"Events generated: {len(events)}")
+
+    report.set_project_state("All", "Event Ingestion")
+    with PerfTimer() as timer:
+        assert usage_extractor, table_refs  # TODO: Replace with call to usage extractor
```
[the remark this answers is not in the document]
Comment (asikowitz) [reply metadata-ingestion/tests/performance/test_bigquery_usage.py:88]: Yeah, but I wrote this to work with the cross-project usage implementation that uses FileBackedDict, or at least the in-memory cross-project usage implementation that I have living in a branch. I don't think the current bigquery code is set up to take in a list of events and `table_refs` like those 2 are, so just leaving the call out here. Then I can add it in in those other branches, once I merge these changes in

**Comment by asikowitz** [comment]
Added changes to metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_report.py that I forgot to add before

File: metadata-ingestion/tests/performance/test_bigquery_usage.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+)
+from datahub.ingestion.source.bigquery_v2.bigquery_report import (
+    BigQueryV2Report,
+    logger as report_logger,
+)
+from datahub.ingestion.source.bigquery_v2.usage import BigQueryUsageExtractor
+from datahub.utilities.perf_timer import PerfTimer
+
+
+def set_log_level(logger: logging.Logger, level: int) -> Generator[None, None, None]:
```
Comment (treff7es) [inline metadata-ingestion/tests/performance/test_bigquery_usage.py:26]: Why is this needed?

Comment (asikowitz) [reply metadata-ingestion/tests/performance/test_bigquery_usage.py:26]: See comment above

File: metadata-ingestion/tests/performance/README.md
Comment (hsheth2) [inline metadata-ingestion/tests/performance/README.md:1]: we'll probably need to add a rule to exclude this file from our docs site

**Review by hsheth2 — APPROVED** [review]
